src/stave.js

```diff
@@ -319,14 +298,96 @@ Vex.Flow.Stave = (function() {
 
     addEndTimeSignature: function(timeSpec, customPadding) {
       this.addEndModifier(new Vex.Flow.TimeSignature(timeSpec, customPadding));
+      return this;
     },
 
     addTrebleGlyph: function() {
-      this.clef = "treble";
-      this.addGlyph(new Vex.Flow.Glyph("v83", 40));
+      this.addClef('treble');
       return this;
     },
 
+    getModifiers: function(position) {
+      if (position === undefined) return this.modifiers;
+
+      return this.modifiers.filter(function(modifier) {
+        return modifier.getPosition() === position;
+      });
+    },
+
+    sortByCategory: function(items, order) {
+      for (var i = items.length - 1; i >= 0; i--) {
+        for (var j = 0; j < i; j++) {
+          if (order[items[j].getCategory()] > order[items[j + 1].getCategory()]) {
+            var temp = items[j];
+            items[j] = items[j + 1];
+            items[j + 1] = temp;
+          }
+        }
+      }
+    },
+
+    format: function() {
```

Comment:
I had to write more details, thanks for good catching. I agree with you and it's better to be removed in the future, but for now I needed to use bubble sort not to break existing order between same category items.

I think Stave is better to allow only one clef, time, key and barline for each BEGIN and END positions. if someone wants to render exceptional special cases, they can use TimeSigNote or ClefNote, etc.
Currently, there are many test cases which add more than one BEGIN or END stavemodifiers into a single stave. I didn't want to break and modify existing test cases this time so that was the reason that I made that kind of custom bubble sort function.

How do you think of preventing adding more than one stavemodifiers which has BEGIN or END positions?

Current solution I thought is this,
1. Updating stave's handling modifiers not to allow multiple same position(BEGIN or END) && category stavemodifiers, during this update I can remove that sortByCategory function.
2. Adding setTimeSignature, setClef, etc(renaming from adding functions, but I will keep existing add functions for compatibility)
3. Updating all tests which adding more than one that kind of stavemodifiers into a single stave, splitting into multiple staves may be good way to change. targets will be test cases like these,
<img width="752" alt="screen shot 2016-01-19 at 4 37 38 pm" src="https://cloud.githubusercontent.com/assets/2025065/12412578/15b2ce48-becb-11e5-91d7-410f7fce22f9.png">
<img width="537" alt="screen shot 2016-01-19 at 4 37 58 pm" src="https://cloud.githubusercontent.com/assets/2025065/12412579/15b491d8-becb-11e5-8caa-7ecc69a9234a.png">